cdap-data-fabric/src/test/java/io/cdap/cdap/spi/data/sql/SqlStructuredTableRetryTest.java

```diff
@@ -58,4 +61,22 @@ protected TransactionRunner getTransactionRunner() {
     Mockito.when((mockedRetryingSqlTransactionRunner).getSqlTransactionRunner()).thenReturn(mockedSqlTransactionRunner);
     return mockedRetryingSqlTransactionRunner;
   }
+
+  @Test
+  public void testSQLExceptionPropagation() throws TransactionException {
+    TransactionRunner transactionRunner = super.getTransactionRunner();
+    SqlTransactionRunner sqlTransactionRunner =
+        ((RetryingSqlTransactionRunner) transactionRunner).getSqlTransactionRunner();
+    try {
+      sqlTransactionRunner.run(
+          context -> {
+            throw new RuntimeException(SQL_CONNECTION_EXCEPTION_MESSAGE,
+                new IOException(SQL_CONNECTION_EXCEPTION_MESSAGE,
+                    new SQLException(SQL_CONNECTION_EXCEPTION, CONNECTION_FAILURE_SQL_STATES)));
```

Comment:
Please add a test for exactly the problem we had: when an exception is thrown from the aux method - `setTransactionIsolation`. Deep test is great to have, but we should also test the specific problem we encountered

Reply:
ack.